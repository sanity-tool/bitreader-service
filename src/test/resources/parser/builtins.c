static float FR, F0, F1;
static float DR, D0, D1;
static long double LDR, LD0, LD1;

static void testFloatBuiltinOps() {
  FR = __builtin_fmodf(F0,F1);
  DR = __builtin_fmod(D0,D1);
  LDR = __builtin_fmodl(LD0,LD1);
}
