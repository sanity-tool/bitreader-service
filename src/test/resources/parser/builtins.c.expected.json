{
  "functions" : [ {
    "ref" : {
      "kind" : "Global",
      "index" : 11
    },
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "typeId" : 1,
        "operands" : [ {
          "kind" : "Global",
          "index" : 1
        } ],
        "sourceRef" : 0
      }, {
        "kind" : "LLVMLoad",
        "typeId" : 1,
        "operands" : [ {
          "kind" : "Global",
          "index" : 2
        } ],
        "sourceRef" : 0
      }, {
        "kind" : "LLVMFRem",
        "typeId" : 1,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 0
        }, {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 1
        } ],
        "sourceRef" : 0
      }, {
        "kind" : "LLVMStore",
        "typeId" : 6,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 2
        }, {
          "kind" : "Global",
          "index" : 3
        } ],
        "sourceRef" : 0
      }, {
        "kind" : "LLVMLoad",
        "typeId" : 1,
        "operands" : [ {
          "kind" : "Global",
          "index" : 4
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMFPExt",
        "typeId" : 7,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 4
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMLoad",
        "typeId" : 1,
        "operands" : [ {
          "kind" : "Global",
          "index" : 5
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMFPExt",
        "typeId" : 7,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 6
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMFRem",
        "typeId" : 7,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 5
        }, {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 7
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMFPTrunc",
        "typeId" : 1,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 8
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMStore",
        "typeId" : 6,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 9
        }, {
          "kind" : "Global",
          "index" : 6
        } ],
        "sourceRef" : 1
      }, {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 8
        } ],
        "sourceRef" : 2
      }, {
        "kind" : "LLVMLoad",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Global",
          "index" : 9
        } ],
        "sourceRef" : 2
      }, {
        "kind" : "LLVMFRem",
        "typeId" : 3,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 11
        }, {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 12
        } ],
        "sourceRef" : 2
      }, {
        "kind" : "LLVMStore",
        "typeId" : 6,
        "operands" : [ {
          "kind" : "Instruction",
          "blockIndex" : 0,
          "index" : 13
        }, {
          "kind" : "Global",
          "index" : 10
        } ],
        "sourceRef" : 2
      }, {
        "kind" : "LLVMRet",
        "typeId" : 6,
        "sourceRef" : 3
      } ]
    } ],
    "entryBlockIndex" : 0
  } ],
  "types" : [ {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 1 ]
  }, {
    "kind" : "LLVMFloatTypeKind"
  }, {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 3 ]
  }, {
    "kind" : "LLVMX86_FP80TypeKind"
  }, {
    "kind" : "LLVMPointerTypeKind",
    "types" : [ 5 ]
  }, {
    "kind" : "LLVMFunctionTypeKind",
    "types" : [ 6 ]
  }, {
    "kind" : "LLVMVoidTypeKind"
  }, {
    "kind" : "LLVMDoubleTypeKind"
  } ],
  "globals" : [ {
    "kind" : "LLVMConstantFPValueKind",
    "typeId" : 1,
    "fpValue" : 0.0
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "F0",
    "typeId" : 0,
    "operands" : [ {
      "kind" : "Global",
      "index" : 0
    } ]
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "F1",
    "typeId" : 0,
    "operands" : [ {
      "kind" : "Global",
      "index" : 0
    } ]
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "FR",
    "typeId" : 0,
    "operands" : [ {
      "kind" : "Global",
      "index" : 0
    } ]
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "D0",
    "typeId" : 0,
    "operands" : [ {
      "kind" : "Global",
      "index" : 0
    } ]
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "D1",
    "typeId" : 0,
    "operands" : [ {
      "kind" : "Global",
      "index" : 0
    } ]
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "DR",
    "typeId" : 0,
    "operands" : [ {
      "kind" : "Global",
      "index" : 0
    } ]
  }, {
    "kind" : "LLVMConstantFPValueKind",
    "typeId" : 3,
    "fpValue" : 0.0
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "LD0",
    "typeId" : 2,
    "operands" : [ {
      "kind" : "Global",
      "index" : 7
    } ]
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "LD1",
    "typeId" : 2,
    "operands" : [ {
      "kind" : "Global",
      "index" : 7
    } ]
  }, {
    "kind" : "LLVMGlobalVariableValueKind",
    "name" : "LDR",
    "typeId" : 2,
    "operands" : [ {
      "kind" : "Global",
      "index" : 7
    } ]
  }, {
    "kind" : "LLVMFunctionValueKind",
    "name" : "testFloatBuiltinOps",
    "typeId" : 4
  } ],
  "sourceRefs" : [ {
    "file" : "parser/builtins.c",
    "line" : 6
  }, {
    "file" : "parser/builtins.c",
    "line" : 7
  }, {
    "file" : "parser/builtins.c",
    "line" : 8
  }, {
    "file" : "parser/builtins.c",
    "line" : 9
  } ]
}